internal/oci/runtime_vm.go

```diff
@@ -52,6 +53,33 @@ type containerInfo struct {
 	cio *cio.ContainerIO
 }
 
+func buildShimV2Path(path string) string {
+	// shim v2 expects the runtime name to be in the following version:
+	//        ($dir.)?$prefix.$name.$version
+	//        -------- ------ ----- ---------
+	//              |     |     |      |
+	//              v     |     |      |
+	//      "/usr/local/bin"    |      |
+	//        (optional)  |     |      |
+	//                    v     |      |
+	//             "containerd.shim."  |
+	//                          |      |
+	//                          v      |
+	//                     "kata-qemu" |
+	//                                 v
+	//                                "v2"
+	const expectedPrefix = "containerd-shim-"
+	const expectedVersion = "-v2"
+
+	const binaryPrefix = "containerd.shim"
+	const binaryVersion = "v2"
+
+	runtimeDir := filepath.Dir(path)
+	runtimeName := strings.SplitAfter(strings.Split(filepath.Base(path), expectedVersion)[0], expectedPrefix)[1]
+
+	return filepath.Join(runtimeDir, fmt.Sprintf("%s.%s.%s", binaryPrefix, runtimeName, binaryVersion))
```

Comment:
Here we enforce the containerd v2 shim prefix and suffix (also if not present in the passed path).

Reply:
do we verify the file exists later?

Reply:
We, as CRI-O, don't, but do we need?
Once this name is built it's passed to https://github.com/containerd/containerd/blob/08d765af954cd5355b7207fe628d06ed5a97c9d4/runtime/v2/shim/util.go#L41, which does the validation here https://github.com/containerd/containerd/blob/08d765af954cd5355b7207fe628d06ed5a97c9d4/runtime/v2/shim/util.go#L61-L94

Reply:
Oh, actually CRI-O does check the existence of the binary during the validation step.